minique/work/worker.py

```diff
@@ -12,31 +15,31 @@ class Worker:
     queue_timeout = 1
     job_runner_class = JobRunner
 
-    def __init__(self, redis, queues):
+    def __init__(self, redis: Redis, queues: List[Queue]) -> None:
         self.id = self.compute_id()
         self.redis = redis
-        self.queues = queues
+        self.queues = list(queues)
```

Comment:
Is this conversion necessary id it is defined as a list in the params?

Reply:
Yes – parameter annotations in Python are just annotations that don't enforce types.
Secondly, this makes sure we "own" the list (as we're grabbing a copy of it), and external modifications to it (if user code is... misinformed enough to do so) won't affect this Worker.

If the annotation was `Iterable[Queue]` (which it could well be), this listification would be mandatory, as iterables such as generators are generally iterable only once.